src/main/java/com/twilio/http/Request.java

```diff
@@ -175,9 +176,21 @@ private String buildURL() {
                 host = joinIgnoreNull(".", product, targetEdge, targetRegion, domain);
             }
 
-            return new URI(parsedUrl.getProtocol(), parsedUrl.getUserInfo(), host, parsedUrl.getPort(),
-                    parsedUrl.getPath(), parsedUrl.getQuery(), parsedUrl.getRef()).toURL().toString();
-        } catch (final MalformedURLException | URISyntaxException e) {
+            String urlPort = null;
+            if (parsedUrl.getPort() != -1) {
+                urlPort = ":" + parsedUrl.getPort();
+            }
```

Comment:
Could do one more simplification here similar to below:
`String urlPort = parsedUrl.getPort() != -1 ? ":" + parsedUrl.getPort() : null;`